www/js/app.js

```diff
@@ -1650,6 +1669,27 @@ function archiveReadyCallback (archive) {
             params.originalContentInjectionMode = null;
         }
     }
+    if (settingsStore.getItem('trustedZimFiles') === null) {
+        settingsStore.setItem('trustedZimFiles', '', Infinity);
+    }
+    if (params.sourceVerification && params.contentInjectionMode === 'serviceworker'){
+        // Check if source of the zim file can be trusted.
+        if (!(settingsStore.getItem('trustedZimFiles').includes(archive.file.name))) {
+            // Alert user about unknown source.
+            console.log("check22")
+            const response = await uiUtil.systemAlert('Do you trust this source?\n You can still read the zim file in Safe Mode incase you cannot verify the source. Closing this window also opens the file in Safe Mode. This option can be disabled in Expert Settings', 'Alert!', true, 'Open in Safe Mode', 'Trust Source')
```

Comment:
As this will be inserted as HTML, it's no good using newline codes, they'll be ignored. You should instead wrap each separate paragraph in `<p>...</p>` inside this string.

Some stylistic suggestions:

```suggestion
            const response = await uiUtil.systemAlert('Is this ZIM archive from a trusted source?\n If not, you can still read the ZIM file in Safe Mode (aka JQuery mode). Closing this window also opens the file in Safe Mode. This option can be disabled in Expert Settings', 'Security alert!', true, 'Open in Safe Mode', 'Trust Source');
```